actuator/src/main/java/org/tron/core/vm/nativecontract/TokenIssueProcessor.java

```diff
@@ -0,0 +1,84 @@
+package org.tron.core.vm.nativecontract;
+
+import org.tron.common.utils.ByteArray;
+import org.tron.common.utils.DecodeUtil;
+import org.tron.core.capsule.AccountCapsule;
+import org.tron.core.capsule.AssetIssueCapsule;
+import org.tron.core.exception.ContractValidateException;
+import org.tron.core.utils.TransactionUtil;
+import org.tron.core.vm.nativecontract.param.TokenIssueParam;
+import org.tron.core.vm.repository.Repository;
+
+import java.util.Objects;
+
+import static org.tron.core.vm.nativecontract.ContractProcessorConstant.*;
+
+public class TokenIssueProcessor {
+
+    public void execute(Object contract, Repository repository) {
+        TokenIssueParam tokenIssueParam = (TokenIssueParam) contract;
+        long tokenIdNum = repository.getTokenIdNum();
+        tokenIdNum++;
+        repository.saveTokenIdNum(tokenIdNum);
+        AssetIssueCapsule assetIssueCapsule = new AssetIssueCapsule(tokenIssueParam.getOwnerAddress(),
+                Long.toString(tokenIdNum), ByteArray.toStr(tokenIssueParam.getName()),
+                ByteArray.toStr(tokenIssueParam.getAbbr()), tokenIssueParam.getTotalSupply(),
+                tokenIssueParam.getPrecision());
+        repository.putAssetIssueValue(assetIssueCapsule.createDbV2Key(), assetIssueCapsule);
+        AccountCapsule accountCapsule = repository.getAccount(tokenIssueParam.getOwnerAddress());
+        accountCapsule.setAssetIssuedName(assetIssueCapsule.getName().toByteArray());
+        accountCapsule.setAssetIssuedID(ByteArray.fromString(assetIssueCapsule.getId()));
+        accountCapsule
+                .addAssetV2(assetIssueCapsule.createDbV2Key(), tokenIssueParam.getTotalSupply());
+        accountCapsule.setInstance(accountCapsule.getInstance().toBuilder().build());
+        // spend 1024trx for assetissue, send to blackhole address
+        AccountCapsule bhAccountCapsule = repository.getAccount(repository.getBlackHoleAddress());
+        bhAccountCapsule.setBalance(Math.addExact(bhAccountCapsule.getBalance(), TOKEN_ISSUE_FEE));
```

Comment:
`TOKEN_ISSUE_FEE` is not a constant. It's a chain parameter in dynamicProperty and is modified through proposals.

Reply:
Good point. Will refine this code in next release due to code freezing :)